modules/core/src/nan_mask.simd.hpp

```diff
@@ -0,0 +1,416 @@
+    for(; i <= (int)total - npixels; i += npixels )
+    {
+        v_uint32 vv = v_ne(v_and(vx_load(src + i*4), vmaskExp), vmaskExp);
+        v_uint8 velems = v_pack_b(vv, z, z, z);
+
+        v_uint32 vresWide = v_eq(v_reinterpret_as_u32(velems), vmaskAll4);
+
+        v_uint8 vres = v_pack_b(vresWide, z, z, z);
+        if (npixels == 1) // 128 bit wide
+        {
+            dst[i] = v_get0(vres);
+        }
+        else if (npixels == 2) // 256 bit wide
+        {
+            *((uint16_t*)(dst + i)) = v_get0(v_reinterpret_as_u16(vres));
+        }
+        else if (npixels == 4) // 512 bit wide
+        {
+            *((uint32_t*)(dst + i)) = v_get0(v_reinterpret_as_u32(vres));
+        }
+        else if (npixels == 8) // 1024 bit wide
+        {
+            *((uint64_t*)(dst + i)) = v_get0(v_reinterpret_as_u64(vres));
+        }
+    }
+    vx_cleanup();
+
+    return i;
+}
+
+
+template <>
+int finiteMaskSIMD_<double, 1>(const double *dsrc, uchar *dst, size_t total)
+{
+    const uint64_t* src = (const uint64_t*)dsrc;
+    const int size8 = VTraits<v_uint8>::vlanes();
+    int i = 0;
+
+    v_uint64 vmaskExp = vx_setall_u64(0x7ff0000000000000);
+    v_uint64 z = vx_setzero_u64();
+    
+    for(; i <= (int)total - (size8 / 2); i += (size8 / 2) )
+    {
+        v_uint64 vv0, vv1, vv2, vv3;
+        vv0 = v_ne(v_and(vx_load(src + i                ), vmaskExp), vmaskExp);
+        vv1 = v_ne(v_and(vx_load(src + i +   (size8 / 8)), vmaskExp), vmaskExp);
+        vv2 = v_ne(v_and(vx_load(src + i + 2*(size8 / 8)), vmaskExp), vmaskExp);
+        vv3 = v_ne(v_and(vx_load(src + i + 3*(size8 / 8)), vmaskExp), vmaskExp);
+
+        v_uint8 v = v_pack_b(vv0, vv1, vv2, vv3, z, z, z, z);
+
+        v_store_low(dst + i, v);
+    }
+    vx_cleanup();
+
+    return i;
+}
+
+template <>
+int finiteMaskSIMD_<double, 2>(const double *dsrc, uchar *dst, size_t total)
+{
+    const uint64_t* src = (const uint64_t*)dsrc;
+    const int size8 = VTraits<v_uint8>::vlanes();
+    const int npixels = size8 / 16;
+    v_uint64 vmaskExp = vx_setall_u64(0x7ff0000000000000);
+    v_uint64 z = vx_setzero_u64();
+
+    int i = 0;
+    for(; i <= (int)total - npixels; i += npixels )
+    {
+        v_uint64 vv = v_ne(v_and(vx_load(src + i*2), vmaskExp), vmaskExp);
+
+        v_uint8 velems = v_pack_b(vv, z, z, z, z, z, z, z);
+
+        v_uint16 vmaskBoth = vx_setall_u16(0xffff);
+        v_uint16 vfinite = v_eq(v_reinterpret_as_u16(velems), vmaskBoth);
+
+        // 2nd arg is useless
+        v_uint8 vres = v_pack(vfinite, vfinite);
+
+        for (int j = 0; j < npixels; j++)
+        {
+            dst[i+j] = v_get0(vres);
+            // 2nd arg is useless
+            vres = v_extract<1>(vres, vres);
+        }
+    }
+    vx_cleanup();
+
+    return i;
+}
+
+
+template <>
+int finiteMaskSIMD_<double, 3>(const double *fsrc, uchar *dst, size_t total)
+{
+    const uint64_t* src = (const uint64_t*)fsrc;
+    const int npixels = VTraits<v_float64>::vlanes();
```

Comment:
Shouldn't this block and related be guarded by `CV_SIMD_64F || CV_SIMD_SCALABLE_64F`?

Reply:
Don't think we need it; all processing is actually done in 64-bit integers